src/views/virtualmachines/utils/virtualMachineStatuses.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+
+import {
+  ExclamationCircleIcon,
+  HourglassHalfIcon,
+  InProgressIcon,
+  OffIcon,
+  PausedIcon,
+  SyncAltIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+
+export const printableVMStatus = {
+  Stopped: 'Stopped',
+  Migrating: 'Migrating',
+  Provisioning: 'Provisioning',
+  Starting: 'Starting',
+  Running: 'Running',
+  Paused: 'Paused',
+  Stopping: 'Stopping',
+  Terminating: 'Terminating',
+  Unknown: 'Unknown',
+};
+
+export const getVMStatusIcon = (
+  status: string,
+  //   arePendingChanges: boolean,
+): React.ComponentClass | React.FC => {
+  let icon: React.ComponentClass | React.FC = UnknownIcon;
+
+  if (status === printableVMStatus.Paused) {
+    icon = PausedIcon;
+  } else if (status === printableVMStatus.Running) {
+    icon = SyncAltIcon;
+  } else if (status === printableVMStatus.Stopped) {
+    icon = OffIcon;
+  } else if (status.toLowerCase().includes('error')) {
+    icon = ExclamationCircleIcon;
+  } else if (
+    status.toLowerCase().includes('pending') ||
+    status.toLowerCase().includes('provisioning')
+  ) {
+    // should be called before inProgress
+    icon = HourglassHalfIcon;
+  } else if (status.toLowerCase().includes('starting')) {
+    icon = InProgressIcon;
+  }
+
+  //   if (arePendingChanges) {
+  //     icon = YellowExclamationTriangleIcon;
```

Comment:
i think we have a new component in our shared library for it.

Reply:
will do :)

Reply:
will refactor once kubevirt-components will be available to this project